python/tests/test_transforms.py

```diff
@@ -318,3 +441,43 @@ def test_void_transform_str():
 
 def test_void_transform_repr():
     assert repr(VoidTransform()) == "VoidTransform()"
+
+
+@pytest.mark.parametrize(
+    "transform,json",
+    [
+        (YearTransform(), '"year"'),
+        (MonthTransform(), '"month"'),
+        (DayTransform(), '"day"'),
+        (HourTransform(), '"hour"'),
+    ],
+)
+def test_datetime_transform_serde(transform, json):
+    assert transform.json() == json
+    assert TestType.parse_raw(json).__root__ == transform
```

Comment:
This is too generic. Each class should be tested individual to validate that parsing the string results in an instance of the correct transform.

Reply:
Split this into different serialize and deserialize tests.